qutebrowser/mainwindow/statusbar/settings.py

```diff
@@ -0,0 +1,130 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Generic settings status displayed in the statusbar."""
+
+from qutebrowser.config import config, configtypes
+from qutebrowser.mainwindow.statusbar import textbase
+from qutebrowser.utils import objreg, message
+
+
+class BooleanSettings(textbase.TextBase):
+
+    """Boolean settings state displayed in the statusbar."""
+
+    # A list of indicator, setting_name pairs.
+    _config = []
+    config_option = 'settings.widget.content'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.parse_config()
+
+    def _test_feature(self, setting_name, tab=None, default=False):
+        if not tab:
+            tab = self._current_tab()
+        if not tab:
+            return default
+        try:
+            return tab.test_feature(setting_name)
+        except KeyError:
+            return default
+
+    def _to_bool(self, setting_name, url):
+        """Return a bool for Bool and BoolAsk settings."""
+        try:
+            opt = config.instance.get_opt(setting_name)
+        except config.configexc.Error as err:
+            raise ValueError(str(err))
+        if not opt.supports_pattern:
+            url = None
+        obj = config.instance.get_obj(setting_name, url=url)
+        if isinstance(opt.typ, configtypes.BoolAsk):
+            default = False if obj == 'ask' else obj
+            return self._test_feature(setting_name, default=default)
+        elif isinstance(opt.typ, configtypes.Bool):
+            return obj
+        raise ValueError(
+            "Setting '{}' is not a boolean setting."
+            .format(setting_name)
+        )
+
+    def parse_config(self):
+        """Parse settings from the config option.
+
+        A ValueError will be raised for any errors in parsing or
+        applying the settings.
+        """
+        raw = config.instance.get(self.config_option)
+        config_ = [x.split('|') for x in raw if x.strip()]
+        # Use `_to_bool()` to do some early validation
+        for part in config_:
+            try:
+                _indicater, value = part
+                self._to_bool(value, None)
+            except ValueError as err:
+                message.error(
+                    "Error parsing '{}' setting value {}: {}"
+                    .format(self.config_option, part, err)
+                )
+                return
```

Comment:
Like above, this kind of validation should probably be the config's responsibility.